vividus-plugin-aws-s3/src/main/java/org/vividus/aws/s3/steps/S3BucketSteps.java

```diff
@@ -55,14 +57,39 @@
      * @param contentType Mime type of object for upload (see <a href="https://en.wikipedia.org/wiki/MIME">MIME</a>)
      * @param bucketName S3 bucket to upload
      */
-    @When("I upload `$resourcePath` with key `$objectKey` and content type `$contentType` to S3 bucket `$bucketName`")
+    @When("I upload resource `$resourcePath` with key `$objectKey` and content type `$contentType`"
+            + " to S3 bucket `$bucketName`")
     public void uploadResource(String resourcePath, String objectKey, String contentType, String bucketName)
     {
         byte[] resource = ResourceUtils.loadResourceAsByteArray(resourcePath);
+        uploadContent(bucketName, objectKey, resource, contentType);
+    }
+
+    /**
+     * Uploads <b>file</b> into S3 given bucket by the <b>objectKey</b>
+     * <br>
+     * Usage example:
+     * <code><br>When I upload file`C:/Users/user/Temp/test.csv` with key `folder/name.csv`
+     *  and content type `text/csv` to S3 bucket `testBucket`</code>
+     * @param file File for upload
+     * @param objectKey Key on which the content is added to S3 bucket
+     * @param contentType Mime type of object for upload (see <a href="https://en.wikipedia.org/wiki/MIME">MIME</a>)
+     * @param bucketName S3 bucket to upload
+     * @throws IOException in case of error on file reading
+     */
+    @When("I upload `$file` with key `$objectKey` and content type `$contentType` to S3 bucket `$bucketName`")
+    public void uploadFile(File file, String objectKey, String contentType, String bucketName) throws IOException
```

Comment:
btw do we really need two steps? we can distinguish between fs files and class path resources by their existence or even use file: or classpath: prefixes to clarify their type

Reply:
for end users it might be difficult to understand the difference